grails-app/controllers/com/careerfly/social/DiscussController.groovy

```diff
@@ -1,64 +1,296 @@
 package com.careerfly.social
 
-import com.careerfly.social.Discussion
-import java.text.SimpleDateFormat
+import com.careerfly.user.User
 
 class DiscussController {
 
     def index() {
 
-        [viewAll: Discussion.list()]
+
+        println(params.int('newsFeeds'))
+        Integer value = params.int('newsFeeds')
+        if(value==null) {
+            [viewAll: Discussion.list()]
+        } else if (value == 1) {
```

Comment:
You could have used some Java constants instead of numbers. That will make code more readable.